### ISSUE TITLE
Add state that indicates that form was changed

### ISSUE THREAD

**Issue by hactrz** [isprojects/mstform#34]
```
@observer
export class MyForm extends Component {
    constructor(props) {
        super(props);
        this.formState = form.state(o);
    }

    @computed
    get dirty() {
        return !_.isEqual(getSnapshot(this.formState.node), this.initial)
    }

    render() {
        const field = this.formState.field("foo");
        return (
            <React.Fragment>
                <InlineError field={field}>
                    <Input type="text" field={field} />
                </InlineError>
                <Button
                    disabled={!this.dirty}
                >
                    Save
                </Button>
            </React.Fragment>
        );
    }
}
```
can we implement state dirty in mstform?

**Comment by faassen** [comment]
:+1: I'm happy to see such a feature added. I think we should add an isDirty for all the accessors (and FormState). The one for field accessor is the interesting one; the others simply aggegrate and are dirty when a child accessors are dirty. What should FieldAccessor.isDirty do? In your example you compare snapshots, but I think a form also dirty if a raw value is filled in that cannot be successfully validated, even though due to the unsuccessful validation the underlying object never gets updated. But I'm willing to be convinced otherwise; I haven't thought the use cases through yet. What would you want to use this for?

I will note that for the whole form you can implement this with the recent 'update' hook too, which is called whenever a value in the underlying object is successfully updated. Something like this:

```js
let dirty = false;

const formState = form.state(obj, { update: () => { dirty = true});
```

**Comment by Amareis** [comment]
You cannot use update hook because user can return form to the initial state in the end.

**Comment by faassen** [comment]
Good point, I hadn't thought of that.

Could you expand on the use cases for dirty checking? I am trying to figure out whether raw values factor into this, or just the value of the underlying object.

**Comment by Amareis** [comment]
It's simple editing form, which have save button. Button is active only if form are dirty.

**Comment by RickLucassen** [comment]
Better late than never ;)

This has been implemented in version 1.32.0, see: https://github.com/isprojects/mstform#dirty-state
